src/unix/freebsd.c

```diff
@@ -205,33 +204,28 @@ int uv_get_process_title(char* buffer, size_t size) {
 
 
 int uv_resident_set_memory(size_t* rss) {
-  kvm_t *kd = NULL;
-  struct kinfo_proc *kinfo = NULL;
-  pid_t pid;
-  int nprocs;
+  struct kinfo_proc kinfo;
   size_t page_size = getpagesize();
+  size_t size = sizeof(struct kinfo_proc);
```

Comment:
Can you use `sizeof(kinfo)` here and below and split the declarations and assignments while you're here?